python/ray/tests/test_runtime_env_container.py

```diff
@@ -0,0 +1,504 @@
+import sys
+
+import pytest
+
+import ray
+from ray.tests.conftest import *  # noqa
+from ray.tests.conftest_docker import *  # noqa
+from ray.tests.conftest_docker import run_in_container, NESTED_IMAGE_NAME
+
+# Runtime env that points to an image that
+# - is a ray image built from the changes in the current commit
+# - contains a custom file that Ray actors can read from when executing requests
+# See `docker/runtime_env_container/Dockerfile` and the `podman_docker_cluster` fixture
+CONTAINER_SPEC = {
+    "image": NESTED_IMAGE_NAME,
+    "worker_path": "/home/ray/anaconda3/lib/python3.8/site-packages/ray/_private/workers/default_worker.py",  # noqa
+}
+CONTAINER_RUNTIME_ENV = {"container": CONTAINER_SPEC}
+
+
+@pytest.mark.skipif(sys.platform != "linux", reason="Only works on Linux.")
+def test_put_get(podman_docker_cluster):
+    """Test ray.put and ray.get."""
+
+    container_id = podman_docker_cluster
+    put_get_script = """
+import ray
+import numpy as np
+
+@ray.remote(runtime_env={
+    "container": {
+        "image": "rayproject/ray:runtime_env_container_nested",
+        "worker_path": "/home/ray/anaconda3/lib/python3.8/site-packages/ray/_private/workers/default_worker.py", # noqa
+    }
+})
+def create_ref():
+    with open("file.txt") as f:
+        assert f.read().strip() == "helloworldalice"
+
+    ref = ray.put(np.zeros(100_000_000))
+    return ref
+
+wrapped_ref = create_ref.remote()
+ray.get(ray.get(wrapped_ref)) == np.zeros(100_000_000)
+""".strip()
```

Comment:
not particular a fan of these inlined python files.. maybe just put them into a directory and build them into your container image with meaningful names?

and then you can run `python <filename.py>` rather than `-c`.